src/nbl/asset/interchange/CGraphicsPipelineLoaderMTL.cpp

```diff
@@ -78,10 +78,12 @@ void CGraphicsPipelineLoaderMTL::initialize()
     }
 
     // default pipelines
-    auto default_mtl_file = m_assetMgr->getFileSystem()->createMemoryReadFile(DUMMY_MTL_CONTENT, strlen(DUMMY_MTL_CONTENT), "default IrrlichtBAW material");
+    system::ISystem::future_t<core::smart_refctd_ptr<system::CMemoryFile>> future;
+    m_assetMgr->getSystem()->createBufferedFile(future, "default IrrlichtBAW material", system::IFile::ECF_READ);
+    auto default_mtl_file = future.get();
 
     SAssetLoadParams assetLoadParams;
-    auto bundle = loadAsset(default_mtl_file, assetLoadParams, &dfltOver);
+    auto bundle = loadAsset(default_mtl_file.get(), assetLoadParams, &dfltOver);
 
     default_mtl_file->drop();
```

Comment:
dont call drop on smart pointer

Reply:
yeah its a remnant of the old API @sadiuk, creation methods didnt use to return smart pointers, now that they do, you shouldn't need to call drop at all